flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/table/JdbcTablePlanTest.java

```diff
@@ -64,4 +82,14 @@ public void testFilterPushdown() {
         util.verifyExecPlan(
                 "SELECT id, time_col, real_col FROM jdbc WHERE id = 900001 AND time_col <> TIME '11:11:11' OR double_col >= -1000.23");
     }
+
+    @Test
+    /**
+     * Note the join condition is not present in the optimized plan, as it is handled in the JDBC java code,
+     * where it adds the join conditions to the select statement string.
+     */
```

Comment:
it's expected, it is present in ast, that's why there are ast and optimization plan in xml

Reply:
Ok that is interesting, though I wonder if the optimisation plan could be aware of what the connector supports, so it could optimize all the supported pushdowns in the main Flink code. Or are there more variation between the way that different databases do predicate pushdowns and the like.

Reply:
Ok that makes sense, though I wonder if the optimisation plan could be aware of what the connector supports, so it could optimize the pushdowns in a standard way. Or are there more variation between the way that different databases do predicate pushdown or the like.